mailbox/plugin/quota-mailing/src/test/java/org/apache/james/mailbox/quota/mailing/search/QuotaSearcherTest.java

```diff
@@ -0,0 +1,170 @@
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.quota.mailing.search;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.james.core.Domain;
+import org.apache.james.core.User;
+import org.apache.james.eventsourcing.EventId;
+import org.apache.james.eventsourcing.eventstore.EventStore;
+import org.apache.james.mailbox.model.Quota;
+import org.apache.james.mailbox.quota.QuotaCount;
+import org.apache.james.mailbox.quota.QuotaSize;
+import org.apache.james.mailbox.quota.mailing.aggregates.UserQuotaThresholds;
+import org.apache.james.mailbox.quota.mailing.events.QuotaThresholdChangedEvent;
+import org.apache.james.mailbox.quota.mailing.search.QuotaQuery.AndQuery;
+import org.apache.james.mailbox.quota.mailing.search.QuotaQuery.HasDomainQuery;
+import org.apache.james.mailbox.quota.mailing.search.QuotaQuery.LessThanQuery;
+import org.apache.james.mailbox.quota.mailing.search.QuotaQuery.MoreThanQuery;
+import org.apache.james.mailbox.quota.model.HistoryEvolution;
+import org.apache.james.mailbox.quota.model.QuotaThreshold;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+
+@TestInstance(Lifecycle.PER_CLASS)
+public interface QuotaSearcherTest {
+
+    String DOMAIN = "bar.com";
+    String DOMAIN2 = "JoeBarTeam.com";
+    User USER = User.from("foo", Optional.of(DOMAIN));
+    User USER2 = User.from("foo2", Optional.of(DOMAIN));
+    User USER3 = User.from("foo3", Optional.of(DOMAIN));
+    User USER4 = User.from("foo", Optional.of(DOMAIN2));
+    User USER5 = User.from("foo", Optional.of(DOMAIN));
+    Quota<QuotaSize> _1_ON_3_SIZE_QUOTA = Quota.<QuotaSize>builder().used(QuotaSize.size(10)).computedLimit(QuotaSize.size(30)).build();
+    Quota<QuotaSize> _2_ON_3_SIZE_QUOTA = Quota.<QuotaSize>builder().used(QuotaSize.size(20)).computedLimit(QuotaSize.size(30)).build();
+    Quota<QuotaCount> _1_ON_4_COUNT_QUOTA = Quota.<QuotaCount>builder().used(QuotaCount.count(1)).computedLimit(QuotaCount.count(4)).build();
+    Quota<QuotaCount> _2_ON_4_COUNT_QUOTA = Quota.<QuotaCount>builder().used(QuotaCount.count(2)).computedLimit(QuotaCount.count(4)).build();
+    Quota<QuotaCount> _3_ON_4_COUNT_QUOTA = Quota.<QuotaCount>builder().used(QuotaCount.count(3)).computedLimit(QuotaCount.count(4)).build();
+    Instant INSTANT = Instant.ofEpochMilli(45554);
+    QuotaThresholdChangedEvent _1_ON_3_SIZE_EVENT = new QuotaThresholdChangedEvent(
```

Comment:
Why do you connect it to `QuotaThresholdChangedEvent` ?

Please, the mean of implementation (using directly the Store\*Manager, relying on MailboxListeners, or whatever technical choice we can make) **belong to the implementation** and **should not be enforced in the contract test**.

Also that very specific event do NOT cary the information we do care about.

BTW we *may* want the QuotaSearcher to work even without **QuotaMailing**